spyder/app/utils.py

```diff
@@ -77,15 +82,32 @@ def setup_logging(cli_options):
         log_level = levels[get_debug_level()]
         log_format = '%(asctime)s [%(levelname)s] [%(name)s] -> %(message)s'
 
+        handlers = [logging.StreamHandler()]
         if cli_options.debug_output == 'file':
             log_file = 'spyder-debug.log'
+            handlers.append(logging.FileHandler(open(log_file, 'w+')))
         else:
             log_file = None
 
-        logging.basicConfig(level=log_level,
-                            format=log_format,
-                            filename=log_file,
-                            filemode='w+')
+        match_func = lambda x: True
+        if FILTER_NAMES != [''] and len(FILTER_NAMES) > 0:
+            dafsa = DAFSA(FILTER_NAMES)
+            match_func = lambda x: (dafsa.lookup(x, stop_on_prefix=True)
+                                    is not None)
+
+        f = logging.Formatter(log_format)
+
+        class LSPFilter(logging.Filter):
```

Comment:
Why is this class necessary? Please add a docstring to understand its purpose.

Reply:
Also, it's name is not clear to me. Is this related to the LSP? Or does it work as the general filter for our logging messages?

Reply:
Sorry, I left that name when I used it to filter LSP messages, that class is on charge of deciding which messages should be displayed and which not

Reply:
Ok, then it could be called `ModuleFilter`.